libdnf/solv/solver.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_SOLV_SOLVER_HPP
+#define LIBDNF_SOLV_SOLVER_HPP
+
+#include "id_queue.hpp"
+#include "pool.hpp"
+
+#include <filesystem>
+
+extern "C" {
+#include <solv/solver.h>
+}
+
+
+namespace libdnf::solv {
+
+class Solver {
+public:
+    Solver(){};
```

Comment:
I am not sure whether we should have default constructor together with constructor requiring pool.
But it is a minor question. Structure is private, and not initialized Solver class is well described.

Reply:
Yeah, I know. I can try to use unique_ptr<Solver> in GoalPrivate and ModuleGoalPrivate classes to get rid of the the default ctor. Then the uninitialized Solver would be just nullptr.